compiler/src/iree/compiler/Tools/iree_compile_lib.cc

```diff
@@ -134,6 +134,11 @@ int mlir::iree_compiler::runIreecMain(int argc, char **argv) {
         compileToPhases.push_back(name.str());
       });
 
+  llvm::cl::opt<std::string> dumpCompilationPhasesTo(
+      "dump-compilation-phases-to",
+      llvm::cl::desc("Dumps IR at the end of each compilation phase to the "
+                     "given directory."));
```

Comment:
I can follow-up to this with docs here: https://iree.dev/developers/general/developer-tips/#compiling-phase-by-phase